SL/Mailer.pm

```diff
@@ -22,143 +31,122 @@ sub new {
-      $h{notify} = "Disposition-Notification-To: $self->{notify}\n";
-    } else {
-      $h{notify} = "Disposition-Notification-To: $h{from}\n";
-    }
   }
-  
-  print OUT qq|From: $h{from}
-To: $h{to}
-$h{date}$h{'reply-to'}$h{cc}$h{bcc}$h{subject}
-Message-ID: <$msgid>
-$h{notify}X-Mailer: Run my Accounts $self->{version}
-MIME-Version: 1.0
-|;
 
+  my $json = JSON->new->utf8;
 
-  if (@{ $self->{attachments} }) {
-    print OUT qq|Content-Type: multipart/mixed; boundary="$boundary"
+  my $data = {};
 
-|;
-    if ($self->{message} ne "") {
-      print OUT qq|--${boundary}
-Content-Type: $self->{contenttype}; charset="$self->{charset}"
+  $data->{sender}->{name} = $self->{fromname};
+  $data->{sender}->{email} = $self->{from};
 
-$self->{message}
+  if ($self->{replyto}){
+     $data->{replyTo}->{name} = $self->{replyto};
+     $data->{replyTo}->{email} = $self->{replyto};
+  }
 
-|;
-    }
+  $data->{to}->[0]->{name} = $self->{to};
+  $data->{to}->[0]->{email} = $self->{to};
 
-    foreach my $attachment (@{ $self->{attachments} }) {
+  if ($self->{cc}){
+     $data->{cc}->[0]->{name} = $self->{cc};
+     $data->{cc}->[0]->{email} = $self->{cc};
+  }
 
-      my $application = ($attachment =~ /(^\w+$)|\.(html|text|txt|sql)$/) ? "text" : "application";
-      
-      unless (open IN, $attachment) {
-	close(OUT);
-	return "$attachment : $!";
-      }
-
-      binmode(IN);
-      
-      my $filename = $attachment;
-      # strip path
-      $filename =~ s/(.*\/|$self->{fileid})//g;
-      
-      print OUT qq|--${boundary}
-Content-Type: $application/$self->{format}; name="$filename"; charset="$self->{charset}"
-Content-Transfer-Encoding: BASE64
-Content-Disposition: attachment; filename="$filename"\n\n|;
-
-      my $msg = "";
-      while (<IN>) {;
-        $msg .= $_;
-      }
-      print OUT &encode_base64($msg);
-
-      close(IN);
-      
-    }
-    print OUT qq|--${boundary}--\n|;
+  if ($self->{bcc}){
+     $data->{bcc}->[0]->{name} = $self->{bcc};
+     $data->{bcc}->[0]->{email} = $self->{bcc};
+  }
+
+  if (@{$self->{attachments}}) {
+      my $i = 0;
+      foreach my $attachment (@{$self->{attachments}}) {
+          my $filename    = $attachment;
+          $filename =~ s/(.*\/|$self->{fileid})//g;
 
-  } else {
-    print OUT qq|Content-Type: $self->{contenttype}; charset="$self->{charset}"
+          $data->{attachment}->[$i]->{name} = $filename;
+          $data->{attachment}->[$i]->{url} = "$tmpfolder/$attachment";
 
-$self->{message}
-|;
+          $i++;
+    }
   }
 
-  close(OUT);
+  $data->{subject} = $self->{subject};
+  $data->{htmlContent} = $self->{message};
+
+  my $jsonstr = $json->encode($data);
+
+  $commandline = q~
+  curl -s --request POST \
+      --url https://api.sendinblue.com/v3/smtp/email \
+      --header 'accept: application/json' \
+      --header 'api-key:~.$apikey.q~' \
+      --header 'content-type: application/json' \
+      --data '~.$jsonstr.q~' > /dev/null \
+  ~;
```

Comment:
Shall this curl be configurable? Would be cool so we could use any api via curl, couldn't we?

Reply:
Yes, we can make it. But we need to first check if the field names are also same for that API. We can do it at later stage once we have all this working with sendinblue.